django/contrib/auth/forms.py

```diff
@@ -186,7 +186,7 @@ def clean(self):
         username = self.cleaned_data.get('username')
         password = self.cleaned_data.get('password')
 
-        if username and password:
+        if username is not None and password is not None:
```

Comment:
This is probably not that important, but the `password is not None` part is not tested.